### DISCUSSION TITLE
Update Roast Yield in Artisan Plus

### DISCUSSION THREAD

**Discussion by CoffeeWorx** [artisan-roaster-scope/artisan#1522] [Q&A]
I know this must be simple to fix but Artisan doesn't seem to update yield/net weight if we've already started a new batch. I need accurate reports of yield in A+ but if that info is entered after a new roast is started then yield shows as 0 in A+ even if yield is entered into the roast log. 

Does anyone know how to get roast info to update in A+?

Wifi is connected. Artisan Plus is linked. Roast batches are showing. But any info entered AFTER the roast has finished doesn't get updated in A+

**Comment by MAKOMO** [comment]
Not completely simple. Sorry. However, data entered on either side, Artisan or the artisan.plus platform, is automatically synced between the two (but note that only information save to disk on the Artisan side is synced back to artisan.plus). Let me explain a bit more.

The roast properties you edit in Artisan belong always to the current roast profile loaded or being recorded. If you like to change/add information to a previous roast, you need to open that one with Artisan first to edit it. This is like in a text processor. You can only change the text of the actual document loaded, not of one not currently open. In contrast to a text process Artisan is only able to open one roast (document). However, a second instance of Artisan can be started (named ArtisanViewer) which comes with a restricted feature set not allowing to record roasts, but to edit properties of roasts.

Let's assume that you are roasting a session and after having started the next roast you'd like to add the yield of the previous roast just measured. You can ask Artisan to automatically open a just completed roast in ArtisanViewer (a second Artisan app instance) once a new roast is started (menu Config >> Sampling, tick "Open Completed Roast in Viewer"). That ArtisanViewer can as well be connected to artisan.plus and you can edit with it the Roast Properties of previous roasts while the main Artisan is already recording the next roast. Sadly this mechanism to send completed roasts to the ArtisanViewer is broken in Artisan v2.10.2 on Windows, but the just released Artisan 2.10.4 does fix this.

Note that at any time you can change/add information on completed roasts on the artisan.plus platform. Just open plus on your phone or a Web browser, choose the roast you want to add the yield information, click edit and type in the yield. The modified information will be synced back to the .alog profile automatically once opened in Artisan (while being connected to artisan.plus).

Anyhow, thinks for batch roasting will get much easier this summer...
